### PR TITLE
Make CLIP support multiple images per sample

### PR DESCRIPTION
**Pull request by zhiqiangdon**
1. Added multiple images support for CLIP.
2. Made the data loader more robust by handling exceptions in loading one sample. Previously, failure of opening missing images could crash training.
3. Added the choice of using a zero image if an image is missing.
4. Improved init_data_processors. Modalities with empty data processors are dropped.
5. Set to use the default AutoMM logger in preprocess_dataframe.py, process_image.py, and process_text.py.
6. Converted metric names to lower case before calling the metric function.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-1606-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1606/1/index.html

File: text/src/autogluon/text/automm/data/dataset.py [hunk 1]
```diff
@@ -69,8 +75,16 @@ def __getitem__(self, idx):
...
-        for per_modality, per_modality_processors in self.processors.items():
-            for per_model_processor in per_modality_processors:
-                ret.update(per_model_processor(getattr(self, per_modality), idx, self.is_training))
-
+        try:  # TODO: consider handling image exceptions within the process_image.py.
+            # TODO: If no images are open, then throw an exception to be captured here.
+            for per_modality, per_modality_processors in self.processors.items():
+                for per_model_processor in per_modality_processors:
+                    ret.update(per_model_processor(getattr(self, per_modality), idx, self.is_training))
+        except Exception as e:
```
Comment (sxjscience) [inline text/src/autogluon/text/automm/data/dataset.py:83]: Would we increase `self._consecutive_errors += 1` ?

Comment (zhiqiangdon) [reply text/src/autogluon/text/automm/data/dataset.py:83]: Right. Added.

**Review by sxjscience — APPROVED** [review]
LGTM. However, we will need to add another option (e.g., using zeros as the image if its missing).

**Comment by szha** [comment]
Job PR-1606-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1606/2/index.html

**Comment by szha** [comment]
Job PR-1606-3 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1606/3/index.html

File: text/src/autogluon/text/automm/configs/data/default.yaml [hunk 2]
```diff
@@ -1,5 +1,6 @@
 data:
   image:
+    use_zero_img: False  # Whether to use a zero image if an image is missing.
```
Comment (sxjscience) [inline text/src/autogluon/text/automm/configs/data/default.yaml:3]: How about call it `missing_value_strategy`?

**Comment by szha** [comment]
Job PR-1606-4 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1606/4/index.html

File: text/src/autogluon/text/automm/configs/data/default.yaml (hunk 2 above)
Comment (zhiqiangdon) [reply text/src/autogluon/text/automm/configs/data/default.yaml]: Done.

**Comment by szha** [comment]
Job PR-1606-5 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1606/5/index.html

**Pull request merged**